nixos/modules/services/mail/tlsrpt.nix

```
@@ -0,0 +1,324 @@
+              description = ''
+                Contact information embedded into the reports.
+              '';
+            };
+
+            sender_address = mkOption {
+              type = types.str;
+              example = "noreply@example.com";
+              description = ''
+                Sender address used for reports.
+              '';
+            };
+
+            sendmail_script = mkOption {
+              type = with types; nullOr str;
+              default = if config.services.postfix.enable then "sendmail" else null;
+              defaultText = lib.literalExpression ''
+                if any [ config.services.postfix.enable ] then "sendmail" else null
+              '';
+              description = ''
+                Path to a sendmail-compatible executable for delivery reports.
+              '';
+            };
+          };
+        };
+        default = { };
+        description = ''
+          Flags from {manpage}`tlsrpt-reportd(1)` as key-value pairs.
+        '';
+      };
+
+      extraFlags = mkOption {
+        type = with types; listOf str;
+        default = [ ];
+        description = ''
+          List of extra flags to pass to the tlsrpt-reportd executable.
+
+          See {manpage}`tlsrpt-report(1)` for possible flags.
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    environment.etc = {
+      "tlsrpt/collectd.cfg".source = format.generate "tlsrpt-collectd.cfg" {
+        tlsrpt_collectd = dropNullValues cfg.collectd.settings;
+      };
+      "tlsrpt/fetcher.cfg".source = format.generate "tlsrpt-fetcher.cfg" {
+        tlsrpt_fetcher = dropNullValues cfg.fetcher.settings;
+      };
+      "tlsrpt/reportd.cfg".source = format.generate "tlsrpt-reportd.cfg" {
+        tlsrpt_reportd = dropNullValues cfg.reportd.settings;
+      };
+    };
+
+    systemd.services.postfix.serviceConfig.SupplementaryGroups = mkIf (
+      config.services.postfix.enable && cfg.collectd.configurePostfix
+    ) [ "tlsrpt" ];
+
+    systemd.services.tlsrpt-collectd = {
+      description = "TLSRPT datagram collector";
+      documentation = [ "man:tlsrpt-collectd(1)" ];
+
+      wantedBy = [ "multi-user.target" ];
+
+      restartTriggers = [ "/etc/tlsrpt/collectd.cfg" ];
+
+      serviceConfig = commonServiceSettings // {
+        ExecStart = toString (
+          [
+            (lib.getExe' cfg.package "tlsrpt-collectd")
+          ]
+          ++ cfg.collectd.extraFlags
+        );
+        IPAddressDeny = "any";
+        PrivateNetwork = true;
+        RestrictAddressFamilies = [ "AF_UNIX" ];
+        RuntimeDirectory = "tlsrpt";
+        RuntimeDirectoryMode = "0750";
+        UMask = "0157";
+      };
+    };
+
+    systemd.services.tlsrpt-reportd = {
+      description = "TLSRPT report generator";
+      documentation = [ "man:tlsrpt-reportd(1)" ];
+
+      wantedBy = [ "multi-user.target" ];
+
+      restartTriggers = [ "/etc/tlsrpt/reportd.cfg" ];
```

```suggestion
      restartTriggers = [ config.environment.etc."tlsrpt/reportd.cfg".source ];
```
I couldn't find any evidence that this would be working. We basically write the contents of this option to a file (https://github.com/NixOS/nixpkgs/blob/7f06b11f32b2c5574c86fc7ed00b97401179965f/nixos/lib/systemd-lib.nix#L586-L592) and trigger a restart when that changes. Supplying a constant path to a file seems not to be handled, hence we would never trigger a restart here. Not sure if the other files should be included as well, didn't look anything up in the upstreams doc/code/etc.